### PR TITLE
First PR to add fork option

### PR DESCRIPTION
**Pull request by xinity**
related to gitter talks, 
i've tried to add a fork option to downstream plugin

### CODE REVIEW

**Comment by tboerger** [comment]
It's still missing go fmt ./...

File: main.go [hunk 1]
```diff
@@ -41,14 +42,21 @@ func main() {
 			fmt.Printf("Error: unable to parse repository name %s.\n", entry)
 			os.Exit(1)
 		}
-
+		if v.fork == "true" {
+		// start a new  build
+		_, err = client.BuildFork(owner, name, build.Number)
```
Comment (tboerger) [inline main.go]: build, err :=
and afterwards you should print a message like in the else at the end

File: main.go [hunk 2]
```diff
@@ -41,14 +42,21 @@ func main() {
 			fmt.Printf("Error: unable to parse repository name %s.\n", entry)
 			os.Exit(1)
 		}
-
+		if v.fork == "true" {
```
Comment (tboerger) [inline main.go]: Just if fork {

Comment (tboerger) [inline main.go]: if v.Fork {

**Comment by tboerger** [comment]
At the end you should run make deps build than you can see errors if there are still some.

File: main.go [hunk 3]
```diff
@@ -41,14 +42,21 @@ func main() {
...
+		// start a new  build
+		_, err = client.BuildFork(owner, name, build.Number)
+			if err != nil {
+				fmt.Printf("Error: unable to trigger a new build for %s.\n", entry)
+				os.Exit(1)
+		}
+		} else {
 		// get the latest build for the specified repository
 		build, err := client.BuildLast(owner, name, branch)
 		if err != nil {
```
Comment (bradrydzewski) [inline main.go]: this needs to happen before `client.BuildFork` so we know which build number to fork from

**Comment by tboerger** [comment]
It's still not properly formatted, go fmt will fix that :)

File: main.go [hunk 4]
```diff
@@ -41,22 +42,36 @@ func main() {
...
+				fmt.Printf("Error: unable to trigger build for %s.\n", entry)
+				os.Exit(1)
+			}
 
-		// rebuild the latest build
-		_, err = client.BuildStart(owner, name, build.Number)
-		if err != nil {
-			fmt.Printf("Error: unable to trigger build for %s.\n", entry)
-			os.Exit(1)
+			fmt.Printf("Restarting build %d for %s\n", build.Number, entry)
```
Comment (tboerger) [inline main.go]: You should add a message like that to the fork part as well.

**Comment by tboerger** [comment]
Got a single fix, afterwards it looks already pretty well. You should squash your commits to keep the history clean.

**Pull request closed without merging**